tests/tensor/test_math.py

```diff
@@ -2076,6 +2079,74 @@ def is_super_shape(var1, var2):
                             assert is_super_shape(y, g)
 
 
+class TestMatrixVectorOps:
+    """Test vecdot, matvec, and vecmat helper functions."""
+
+    @pytest.mark.parametrize(
+        "func,x_shape,y_shape,np_func,batch_axis",
+        [
+            # vecdot
+            (vecdot, (5,), (5,), lambda x, y: np.dot(x, y), None),
```

Comment:
`np_func` just use the numpy equivalents. Skip if numpy < 2.0 because they didn't exist. No need for these lambda x, y, which may be incorrect anyway